src/meta/src/rpc/service/notification_service.rs

```diff
@@ -102,6 +100,8 @@ where
             .filter(|mapping| table_ids.contains(&mapping.table_id))
             .collect_vec();
 
+        let hummock_manager_rdguard = self.hummock_manager.get_read_guard().await;
```

Comment:
IIUC the guard to avoid missing deltas between creating snapshot and `insert_sender`. I see `tables` and `hummock_version` both apply this approach. Should other snapshot fields like `databases`, `schemas`, etc do the same? @yezizp2012

Reply:
> IIUC the guard to avoid missing deltas between creating snapshot and `insert_sender`. I see `tables` and `hummock_version` both apply this approach. Should other snapshot fields like `databases`, `schemas`, etc do the same? @yezizp2012

yes, the guard is set to avoid missing deltas between creating snapshot and `insert_sender`

Reply:
> IIUC the guard to avoid missing deltas between creating snapshot and `insert_sender`. I see `tables` and `hummock_version` both apply this approach. Should other snapshot fields like `databases`, `schemas`, etc do the same? @yezizp2012 

Yes, other fields already applied this approach. For example, `databases`, `schemas` are protected by `catalog_guard` as same as `tables`.